pwiz_tools/Skyline/Model/Irt/IrtDb.cs

```diff
@@ -591,5 +591,23 @@ private static void EnsureDocumentXmlTable(ISession session)
                 }
             }
         }
+
+        public void RemoveLibraryPeptides(IEnumerable<Target> peptides)
+        {
+            using (var session = OpenWriteSession())
+            using (var transaction = session.BeginTransaction())
+            {
+                foreach (var pep in peptides)
+                {
+                    using (var cmd = session.Connection.CreateCommand())
+                    {
+                        cmd.CommandText = @"DELETE FROM IrtLibrary WHERE Standard = 0 AND PeptideModSeq = ?";

```

Comment:
If you want to remove all non-standard peptides which have the same sequence as a standard peptide you could do:
DELETE FROM IrtLibrary L1 WHERE L1.Standard = 0 and L1.PeptideModSeq IN (SELECT L2.PeptideModSeq FROM IrtLibrary L2 WHERE L2.Standard = 1)
Afterwards, if you wanted to make sure that there were no duplicates you could do:
DELETE FROM IrtLibrary L1 WHERE L1.Id <> (SELECT MIN(L2.Id) FROM IrtLibrary L2 WHERE L2.PeptideModeSeq = L1.PeptideModSeq)

Reply:
You're right, that's a better way to do it.